Dingo/app/src/main/java/com/example/dingo/trips/TripViewModel.kt

```diff
@@ -0,0 +1,124 @@
+package com.example.dingo.trips
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.liveData
+import com.example.dingo.model.Location
+import com.example.dingo.model.Trip
+import com.example.dingo.model.User
+import com.example.dingo.model.service.TripService
+import com.example.dingo.model.service.UserService
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.withContext
+import java.util.PriorityQueue
+import javax.inject.Inject
+
+@HiltViewModel
+class TripViewModel
+@Inject
+constructor(
+    private val userService: UserService,
+    private val tripService: TripService,
+) : ViewModel() {
+
+    fun makeTrip(
+        userId: String,
+        username: String,
+        locations: List<Location>,
+        discoveredEntries: List<String>
+    ) {
+        runBlocking{
+            var tripId = withContext(Dispatchers.Default) {
+                tripService.createTrip(
+                    userId,
+                    username,
+                    locations,
+                    discoveredEntries
+                )
+            }
+
+            var user = withContext(Dispatchers.Default) {
+                userService.getUser(userId)
+            }
+
+            if (user != null) {
+//                val currPostHead = user.postHead
+//                if (currPostHead.isNotEmpty()) {
+//                    postService.setPostNext(currPostHead, postId)
+//                }
+//                postService.setPostPrev(postId, currPostHead)
+//                userService.setPostHeadForUser(userId, postId, PostType.SOCIAL_POST)
+            } else {
+                println("Could not find user with user id $userId when making Trip $tripId")
+            }
+
+        }
+    }
+
+    fun getFeedForUser(userId: String, limit: Int = 10): MutableList<Trip> {
+        var ret = mutableListOf<Trip>()
+
+        runBlocking {
+            var user = withContext(Dispatchers.Default) {
+                userService.getUser(userId)
+            }
+
+            if (user != null) {
```

Comment:
It would probably be best to use the classroom feed as a template for seeing a user's own trips, instead of the social feed, since the classroom feed uses an array in the firebase, which is closer to how the trips are stored for each user

Reply:
Thanks @simhonchourasia that is a good idea. Made the changes!